dart/lib/src/event_processor/deduplication_event_processor.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import '../event_processor.dart';
+import '../protocol.dart';
+import '../sentry_options.dart';
+
+const eventsToKeepForDeduplication = 20;
+
+/// Deduplicates events with the same [SentryEvent.throwable].
+/// It keeps track of the last [eventsToKeepForDeduplication]
+/// events. Older events aren't considered for deduplication.
+/// This is a hard
+/// Caveats:
+/// It does not work in the following case:
+/// ```dart
+/// var fooOne = Exception('foo');
+/// var fooTwo = Exception('foo');
+/// ```
+/// because (fooOne == fooTwo) equals false
+class DeduplicationEventProcessor extends EventProcessor {
+  DeduplicationEventProcessor(this._options);
+
+  // Map from exception to object
+  final Map<Object, Object?> exceptionMap = {};
```

Comment:
should we be specific about key and type?

Reply:
The basetype of Exceptions in Dart is Object (the key) because you can throw anything. The value is never used. This is still just a naive conversion of the Java and C# implementations.

Maybe using a [LinkedHashSet](https://api.dart.dev/stable/2.13.3/dart-collection/LinkedHashSet-class.html) is a better option.

Reply:
do we have a thread-safe type? Dart is sort of single-threaded but when using multiple isolates, this could be racey

Reply:
Nope, there's no such thread safe class. Errors should be propagated via `Ports` (see https://api.dart.dev/stable/2.13.3/dart-isolate/dart-isolate-library.html) back to the main isolate. See for example Flutters [`compute` methods source code](https://github.com/flutter/flutter/blob/d79295af24c3ed621c33713ecda14ad196fd9c31/packages/flutter/lib/src/foundation/_isolates_io.dart).

Reply:
nice, so we just need to choose the right data structure for such a case.

Reply:
as we don't want to hold references, nor to leak objects, after discussing, what would make more sense is having a data structure that stores the `hashCode` as an Int, it's way cheaper than the reference of the object in memory.
let's make a circular buffer similar to Scope#breadcrumbs